### PR TITLE
Build standalone dart sdk as part of Flutter engine.

### PR DESCRIPTION
**Pull request by aam**
This builds upon https://codereview.chromium.org/2998503002 to have Dart sdk built as part of Flutter engine builds (targeting host architecture).

Intent is to include this Dart sdk with Flutter engine so that whatever dart script that need to run in support of Dart Frontend can do so on engine-compatible version of Dart sdk.

### CODE REVIEW

File: BUILD.gn [hunk 1]
```diff
@@ -11,6 +11,12 @@ group("flutter") {
     "//flutter/sky",
   ]
 
+  if (current_toolchain == host_toolchain) {
+    deps += [
+        "//dart:create_sdk"
```
Comment (abarth) [inline BUILD.gn]: Do you want to do this for Fuchsia?  I suspect you want to put this in the non-Fuchsia section.

Comment (aam) [inline BUILD.gn]: Yes, and it actually has already been enabled for Fuchsia.

Comment (abarth) [inline BUILD.gn]: In the Fuchsia build, it's not for this BUILD.gn file to decide whether we want to build the Dart SDK.  That's the responsibility of another part of the system (probably something in the build layer).  Think about it this way: if Fuchsia decided to switch to a prebuilt Dart SDK (e.g., similar to how Fuchsia uses a prebuilt llvm toolchain), we shouldn't need to modify this file.

Comment (aam) [inline BUILD.gn]: I think I got it. Thanks!

**Comment by abarth** [comment]
LGTM

**Comment by abarth** [comment]
Thanks!

**Comment by aam** [comment]
@tvolkert , the list is created automatically via ```tools/dart/create_updated_flutter_deps.py``` - it basically pulls dependencies from synced ```dart/DEPS``` file. Produced list has to be manually inserted into ``flutter/DEPS``` at this point though.

**Pull request merged**